### PR TITLE
ui: Error message for non-table data on Databases page

### PR DESCRIPTION
**Pull request by koorosh**
Resolves one part of issue #48152

Previously, error messages received on request for
non-table data on Databases page is not handled and not
displayed for users.

Now, default error message is shown in response to errors.

Release note (admin ui change): Show default error message
about restricted permissions for non-admin users on
Databases page.

Test cases to validate correct behavior:
- Databases page loads table info for a user with `admin` role
- Databases page shows the default error message with `insufficient rights` message for non-admin user.
- The loading indicator is displayed while response is not received.

Before
before

After
after

### CODE REVIEW

File: pkg/ui/src/views/databases/containers/databases/nonTableSummary.tsx
Comment (koorosh) [inline pkg/ui/src/views/databases/containers/databases/nonTableSummary.tsx:45]: It is necessary to preserve the reference to `this` when passing function in props to `Loading` component.

File: pkg/ui/src/views/databases/containers/databases/nonTableSummary.spec.tsx
Comment (celiala) [inline pkg/ui/src/views/databases/containers/databases/nonTableSummary.spec.tsx:26]: nit: remove space after `it`
```
    it("successfully loads...
```

File: pkg/ui/src/views/databases/containers/databases/nonTableSummary.spec.tsx
Comment (celiala) [inline pkg/ui/src/views/databases/containers/databases/nonTableSummary.spec.tsx:54]: nit: same
```javascript
    it("shows error...
```

File: pkg/ui/src/views/databases/containers/databases/nonTableSummary.tsx
Comment (celiala) [inline pkg/ui/src/views/databases/containers/databases/nonTableSummary.tsx:103]: nit: remove spaces
```javascript
            <Loading
              loading={!this.props.nonTableStats}
              error={this.props.lastError}
              render={this.renderTable}
            />
```

**Comment by celiala** [comment]
Hmm, I wonder if we should also hide the "Non-Table Cluster Data" header, since I'm not sure how valuable it it for the user to see that, given that they aren't seeing any data for "Table cluster data". 

But, I'll defer to whatever @piyush-singh and @dhartunian think. Fyi, this PR results in:


But IMO I think it should be without the line and "Non-Table" header, i.e.:

**Review by celiala — CHANGES_REQUESTED** [review]
The code itself LGTM, modulo a few nits (noted inline)...

However, I have a request for the PR itself (this is for all PRs):

* For any UI-facing change, could you include screenshots of the change that you're making, so that it's easy to visually see what area this changes? Ideally a before/after screenshot, if it makes sense. Thanks!

(but let's also hold on the above, to see if Piyush/DavidH think the design should change, of course)

File: pkg/ui/src/views/databases/containers/databases/nonTableSummary.tsx
Comment (koorosh) [reply pkg/ui/src/views/databases/containers/databases/nonTableSummary.tsx:103]: Done

File: pkg/ui/src/views/databases/containers/databases/nonTableSummary.spec.tsx
Comment (koorosh) [reply pkg/ui/src/views/databases/containers/databases/nonTableSummary.spec.tsx]: Done

File: pkg/ui/src/views/databases/containers/databases/nonTableSummary.spec.tsx
Comment (koorosh) [reply pkg/ui/src/views/databases/containers/databases/nonTableSummary.spec.tsx]: Done

File: pkg/ui/src/views/databases/containers/databases/nonTableSummary.spec.tsx
Comment (koorosh) [reply pkg/ui/src/views/databases/containers/databases/nonTableSummary.spec.tsx]: Done

File: pkg/ui/src/views/databases/containers/databases/nonTableSummary.spec.tsx
Comment (koorosh) [reply pkg/ui/src/views/databases/containers/databases/nonTableSummary.spec.tsx]: Done

**Comment by koorosh** [comment]
@celiala thank you for the feedback! Updated code with respect to all requested changes.

**Comment by celiala** [comment]
Updating thread, based on offline (Slack) discussion:

I previously asked (above) whether or not to hide the "Non-Table Cluster Data" header if the "insufficient privileges". This evolved into an investigation around users with various privilege permutations, which evolved into a larger question of what should we show for users with _some_ data privileges.

We landed on keeping this PR/messaging as-is, and create a separate issue (#49782) to account for access to some, but not all, data on Databases page.

**Review by celiala — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
